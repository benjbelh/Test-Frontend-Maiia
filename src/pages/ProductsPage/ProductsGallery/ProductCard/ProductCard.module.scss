.card {
    width: 300px;
    margin: 16px;
    display: flex;
    flex-direction: column;

    .media {
        height: 250px; 
    }

    .cardActions {
        flex-grow: 1;
        align-items: flex-end;
        justify-content: flex-end;
    }
}