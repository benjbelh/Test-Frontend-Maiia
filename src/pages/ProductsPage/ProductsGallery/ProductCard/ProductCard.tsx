import React, { useCallback, FC } from 'react';
import { Button, Card, CardActionArea, CardActions, CardContent, CardMedia, Typography } from '@material-ui/core';
import { useHistory } from 'react-router-dom';
import styles from './ProductCard.module.scss';

type ProductCardProps = {
  id: string;
  name: string;
  picturePath: string;
};

const ProductCard: FC<ProductCardProps> = ({ id = '', name = '', picturePath = '' }) => {
  const history = useHistory();

  const onProductSelected = useCallback(
    (productId) => {
      //TODO IMPLEMENT ADD TO CART
      console.log(productId);
    },
    [history],
  );

  return (
    <Card className={styles.card} data-testid={`product-${id}`}>
      <CardActionArea onClick={() => onProductSelected(id)}>
        {picturePath && <CardMedia className={styles.media} image={picturePath} title={name} />}
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {name}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button size="small" color="primary" onClick={() => onProductSelected(id)}>
          ADD PRODUCT
        </Button>
      </CardActions>
    </Card>
  );
};

export default ProductCard;
