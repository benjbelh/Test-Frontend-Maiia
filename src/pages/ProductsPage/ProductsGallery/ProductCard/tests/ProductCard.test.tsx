import React from 'react';
import { render } from '@testing-library/react';
import ProductCard from '../ProductCard';

describe('ProductCard', () => {
  // it('should renders correctly by default', () => {
  //   const { asFragment } = render(<ProductCard />);
  //   expect(asFragment()).toMatchSnapshot();
  // });
  // it('should renders correctly with card props', () => {
  //   const { asFragment } = render(
  //     <ProductCard id="test-id" name="Product Name" picturePath="https://test.com/img.jpg" location="Cannes" />,
  //   );
  //   expect(asFragment()).toMatchSnapshot();
  // });
});
