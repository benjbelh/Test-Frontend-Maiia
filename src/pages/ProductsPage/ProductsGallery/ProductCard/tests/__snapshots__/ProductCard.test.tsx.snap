// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should renders correctly with card props 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="product-1"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardMedia-root media"
        style="background-image: url(https://test.com/img.jpg);"
        title="Product Name"
      />
      <div
        class="MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Product Name
        </h2>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root cardActions MuiCardActions-spacing"
    >
      <button
        aria-label="add to shopping cart"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
        data-testid="product-1-cart-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
